src/aspire/noise/noise.py

```diff
@@ -183,6 +183,51 @@ def signal_power(self, p):
             self._build()
 
 
+class ColoredNoiseAdder(WhiteNoiseAdder):
+    @abc.abstractmethod
+    def _spectrum(self, x, y):
+        """
+        Colored noise spectrum (2d).
+        """
+
+    def _build(self):
+        """
+        Builds underlying Filter for this NoiseAdder.
+        """
+        custom_filter = FunctionFilter(f=self._spectrum) * ScalarFilter(
+            value=self.noise_var
+        )
+
+        # Call the __init__ from parent of WhiteNoiseAdder.
+        super(WhiteNoiseAdder, self).__init__(
+            noise_filter=custom_filter, seed=self.seed
+        )
+
+
+class BlueNoiseAdder(ColoredNoiseAdder):
+    """
+    NoiseAdder where noise power increases with frequency.
+    """
+
+    def _spectrum(self, x, y):
```

Comment:
Pondering if I should just make `BlueFilter` and `PinkFilter` etc instead (to replace the White's ScalarFilter). However, I'd want to really be sure we have the formulas right if taking those names, and I don't totally understand how it all works in the higher (than 1D) dimensions.